src/watertap_contrib/seto/analysis/net_metering/ro_pv.py

```diff
@@ -0,0 +1,401 @@
+import os
+import numpy as np
+
+from pyomo.environ import (
+    ConcreteModel,
+    Objective,
+    Param,
+    Constraint,
+    Var,
+    Block,
+    TransformationFactory,
+    assert_optimal_termination,
+    units as pyunits,
+)
+from pyomo.network import Arc
+from idaes.core import FlowsheetBlock
+from idaes.core.solvers.get_solver import get_solver
+from idaes.models.unit_models import Product, Feed
+from idaes.core.util.model_statistics import *
+from idaes.core.util.scaling import *
+from idaes.core import UnitModelCostingBlock
+from idaes.core.util.initialization import propagate_state
+from idaes.apps.grid_integration.multiperiod.multiperiod import MultiPeriodModel
+
+from watertap.core.util.initialization import assert_degrees_of_freedom
+from watertap.property_models.NaCl_prop_pack import NaClParameterBlock
+from watertap.unit_models.pressure_changer import Pump, EnergyRecoveryDevice
+
+from watertap.unit_models.reverse_osmosis_0D import (
+    ReverseOsmosis0D,
+    ConcentrationPolarizationType,
+    MassTransferCoefficient,
+    PressureChangeType,
+)
+from watertap.examples.flowsheets.RO_with_energy_recovery.RO_with_energy_recovery import (
+    calculate_operating_pressure,
+)
+from watertap.core.util.infeasible import *
+
+from watertap_contrib.seto.analysis.net_metering.util import *
+from watertap_contrib.seto.costing import (
+    TreatmentCosting,
+    EnergyCosting,
+    SETOSystemCosting,
+)
+from watertap_contrib.seto.solar_models.zero_order import PhotovoltaicZO
+from watertap_contrib.seto.core import SETODatabase, PySAMWaterTAP
+
+
+solver = get_solver()
+
+absolute_path = os.path.dirname(__file__)
+print(absolute_path)
+
+tech_config_file = "/pysam_data/pvsamv1.json"
+tech_config_file = absolute_path + tech_config_file
+grid_config_file = "/pysam_data/grid.json"
+grid_config_file = absolute_path + grid_config_file
+rate_config_file = "/pysam_data/utilityrate5.json"
+rate_config_file = absolute_path + rate_config_file
+cash_config_file = "/pysam_data/singleowner.json"
+cash_config_file = absolute_path + cash_config_file
+weather_file = "/pysam_data/phoenix_az_33.450495_-111.983688_psmv3_60_tmy.csv"
+weather_file = absolute_path + weather_file
+
+
+def build_ro_pv():
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(dynamic=False)
+    m.db = SETODatabase()
```

Comment:
Since I am starting with this PR, I'm unaware what the purpose of `SETODatabase` is. I am guessing you have some cost data/assumptions there but haven't looked into it. Can you tell me a bit about it?

Reply:
At this point it is identical to the ``Database`` class in WaterTAP. I created a separate class initially thinking we would be dealing with different types of data (e.g., location, weather) in SETO, and might need some special functionality at some point. But using the WaterTAP ``Database`` would work in this flowsheet.

The PV model is a ZO model, so requires a database, but the ``PhotovoltaicZO`` model could be changed to a more straightforward structure (e.g. just being a ``Block``). I am open to suggestions.